### ISSUE TITLE
ESLint down due to server crash 

### ISSUE THREAD

**Issue by guppie70** [Microsoft/vscode-eslint#241]
I just updated the ESLint extension to versus 1.2.10 and now suddenly it stopped working completely... Visual Studio code says "The ESLint server crashed 5 times in the last 3 minutes. The server will not be restarted." And that's it...
Tried re-installing the plugin and reinstalling ESLint using: `npm install -g eslint`, but nothing seems to work.
Would love to share more (log) information, but I have not been able to find any. The event viewer does not show any events that seem to relate to an eslint issue. Have no clue where to find log data on my Win 2012 Server R2 Standard machine.
Any help would be greatly appreciated!

**Comment by MarkSMT** [comment]
same here.. on macOS, i updated to the newest vscode version..

**Comment by cecilemuller** [comment]
Adding`"eslint.trace.server": "verbose"` in settings.json should output some information hopefully.

**Comment by guppie70** [comment]
That indeed provides some debug information!

Here is the relevant part I think:
`[Trace - 1:58:08 PM] Sending response 'client/registerCapability - (0)'. Processing request took 2ms
C:\Users\jtadmin\.vscode\extensions\dbaeumer.vscode-eslint-1.2.10\server\server.js:613
    return document2Library[document.uri].then((library) => {
                                         ^

TypeError: Cannot read property 'then' of undefined
    at validateSingle (C:\Users\jtadmin\.vscode\extensions\dbaeumer.vscode-eslint-1.2.10\server\server.js:613:42)
    at Object.messageQueue.onNotification [as handler] (C:\Users\jtadmin\.vscode\extensions\dbaeumer.vscode-eslint-1.2.10\server\server.js:308:5)
    at BufferedMessageQueue.processQueue (C:\Users\jtadmin\.vscode\extensions\dbaeumer.vscode-eslint-1.2.10\server\server.js:297:18)
    at Immediate.timer.setImmediate (C:\Users\jtadmin\.vscode\extensions\dbaeumer.vscode-eslint-1.2.10\server\server.js:260:18)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
[Info  - 1:58:08 PM] Connection to server got closed. Server will restart.
[Info  - 1:58:08 PM] ESLint server stopped.
`

**Comment by cecilemuller** [comment]
That looks like the culprit, good find 😃

**Comment by svipben** [comment]
Same here, this is the output:

```
[Info  - 5:39:05 PM] ESLint server is running.
/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613
    return document2Library[document.uri].then((library) => {
                                         ^

TypeError: Cannot read property 'then' of undefined
    at validateSingle (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613:42)
    at Object.messageQueue.onNotification [as handler] (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:308:5)
    at BufferedMessageQueue.processQueue (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:297:18)
    at Immediate.timer.setImmediate (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:260:18)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
[Info  - 5:39:05 PM] Connection to server got closed. Server will restart.
[Info  - 5:39:05 PM] ESLint server stopped.
[Info  - 5:39:06 PM] ESLint server is running.
/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613
    return document2Library[document.uri].then((library) => {
                                         ^

TypeError: Cannot read property 'then' of undefined
    at validateSingle (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613:42)
    at Object.messageQueue.onNotification [as handler] (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:308:5)
    at BufferedMessageQueue.processQueue (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:297:18)
    at Immediate.timer.setImmediate (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:260:18)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
[Info  - 5:39:06 PM] Connection to server got closed. Server will restart.
[Info  - 5:39:06 PM] ESLint server stopped.
[Info  - 5:39:07 PM] ESLint server is running.
/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613
    return document2Library[document.uri].then((library) => {
                                         ^

TypeError: Cannot read property 'then' of undefined
    at validateSingle (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613:42)
    at Object.messageQueue.onNotification [as handler] (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:308:5)
    at BufferedMessageQueue.processQueue (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:297:18)
    at Immediate.timer.setImmediate (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:260:18)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
[Info  - 5:39:07 PM] Connection to server got closed. Server will restart.
[Info  - 5:39:07 PM] ESLint server stopped.
[Info  - 5:39:08 PM] ESLint server is running.
/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613
    return document2Library[document.uri].then((library) => {
                                         ^

TypeError: Cannot read property 'then' of undefined
    at validateSingle (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613:42)
    at Object.messageQueue.onNotification [as handler] (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:308:5)
    at BufferedMessageQueue.processQueue (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:297:18)
    at Immediate.timer.setImmediate (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:260:18)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
[Info  - 5:39:08 PM] Connection to server got closed. Server will restart.
[Info  - 5:39:08 PM] ESLint server stopped.
[Info  - 5:39:09 PM] ESLint server is running.
/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613
    return document2Library[document.uri].then((library) => {
                                         ^

TypeError: Cannot read property 'then' of undefined
    at validateSingle (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:613:42)
    at Object.messageQueue.onNotification [as handler] (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:308:5)
    at BufferedMessageQueue.processQueue (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:297:18)
    at Immediate.timer.setImmediate (/home/svipben/.vscode/extensions/dbaeumer.vscode-eslint-1.2.10/server/server.js:260:18)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
[Error - 5:39:09 PM] Connection to server got closed. Server will not be restarted.
[Info  - 5:39:09 PM] ESLint server stopped.

```

**Comment by dbaeumer** [comment]
Dups #240. I published 1.2.11. Sorry for the breakage.
